File: .github/skills/codebase-inspector/tests/fixtures/languages/typescript/sample.ts
import { format } from "./format.js";
export interface Runner { run(input: string): Promise<string>; }
export class Greeter implements Runner {
  private prefix: string;
  constructor(prefix: string) { this.prefix = prefix; }
  public async run(input: string = "secret"): Promise<string> { return format(this.prefix + input); }
  static create(): Greeter { return new Greeter("hi"); }
}
export function add(left: number, right: number): number { return left + right; }

File: .github/skills/codebase-inspector/tests/fixtures/languages/rust/sample.rs
use crate::format::format_name;

pub struct Greeter {
    pub prefix: String,
    count: usize,
}

pub enum Status {
    Ready,
    Error(String),
    Named { code: i32 },
}

pub trait Runner {
    fn run(&self, input: &str) -> String;
}

impl Greeter {
    pub fn new(prefix: String) -> Self {
        Self { prefix, count: 0 }
    }

    pub async fn run(&self, input: &str) -> String {
        format_name(input)
    }
}

pub fn add(left: i32, right: i32) -> i32 {
    left + right
}
